sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/key_vault_key.hpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "azure/keyvault/keys/json_web_key.hpp"
+#include "azure/keyvault/keys/key_operation.hpp"
+#include "azure/keyvault/keys/key_properties.hpp"
+
+#include <azure/core/http/http.hpp>
+
+#include <vector>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
+
+  namespace Details {
+    constexpr static const char* KeyPropertyName = "key";
+  } // namespace Details
+
+  struct KeyVaultKey
+  {
+    JsonWebKey Key;
+    KeyProperties Properties;
+
+    KeyVaultKey(std::string const& name) : Properties(name) {}
+
+    std::string const& Id() const { return Key.Id; }
+    std::string const& Name() const { return Properties.Name; }
+    KeyTypeEnum const& GetKeyType() const { return Key.KeyType; }
```

Comment:
Why not just `KeyType` That's what it is in other languages, and seems to fit the pattern I see elsewhere.

Reply:
`JsonWebKey` class has a member of type KeyType. C++ can't handle type and name to be the same for an enum like:

```cpp
enum class KeyType {
    TypeA,
    TypeB
};

struct JsonWebKey {
    KeyType KeyType;
};
```

So I decided to keep the member name as `KeyType` and the enum type name as `KeyTypeEnum`.

Of course names can be updated, but we can't use the same name